spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/JpaProperties.java

```diff
@@ -125,6 +127,22 @@ public void setHibernate(Hibernate hibernate) {
 		return this.hibernate.getAdditionalProperties(this.properties, dataSource);
 	}
 
+	/**
+	 * Return the database form the jdbc url or the value for `spring.jpa.database`.
+	 * @param jdbcUrl the url from `spring.datasource.url`
+	 * @return {@code Database}
+	 */
+	public Database determineDatabase(String jdbcUrl) {
+		Database[] databases = Database.values();
+		for (Database database : databases) {
+			if (database.name().equals(DatabaseDriver.fromJdbcUrl(jdbcUrl)
```

Comment:
You should move that up and back-off if the `DatabaseDriver` is `UNKNOWN`. 

Also, we need a better link between `Database` and `DatabaseDriver`.  when I gave some thought to that issue, I was considering creating a mapping between a `DatabaseDriver` and a `Database`. I think it's better than relying on the fact they share the same name...

We probably need a test that iterates over `Database` values to make sure everything is covered somehow.